packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs

```diff
@@ -904,24 +905,50 @@ where
         let value = serde_json::to_value(entity)
             .into_report()
             .change_context(InsertionError)?;
-        let version = self.as_client().query_one(
-            r#"
-                    INSERT INTO entities (entity_id, version, entity_type_version_id, properties, owned_by_id, created_by_id, updated_by_id)
-                    VALUES ($1, clock_timestamp(), $2, $3, $4, $5, $6)
-                    RETURNING version;
+        let version = self
+            .as_client()
+            .query_one(
+                r#"
+                INSERT INTO entities (
+                    entity_id, version,
+                    entity_type_version_id,
+                    properties,
+                    left_order, right_order,
+                    left_entity_id, right_entity_id,
+                    owned_by_id, created_by_id, updated_by_id
+                )
+                VALUES ($1, clock_timestamp(), $2, $3, $4, $5, $6, $7, $8, $9, $10)
+                RETURNING version;
                 "#,
-            &[&entity_id, &entity_type_version_id, &value, &owned_by_id, &created_by_id, &updated_by_id],
-        )
+                &[
+                    &entity_id,
+                    &entity_type_version_id,
+                    &value,
+                    &link_metadata.as_ref().map(LinkEntityMetadata::left_order),
```

Comment:
At the moment we don't actually check that the entity's entity types include a link type. Which means you could push link metadata for a non-link entity which would be bad.

I'm not sure how we want to handle it right now, we can ignore it for a little while because we don't have any validation but I think this might be a little different to not validating `properties`.

I was thinking we could store `link_type: boolean` in the `entity_types` table, and maybe be able to put a PG condition that the links stuff is null if the entity type isn't a link type? 

Otherwise we can just run a check in the actual graph impl and query the type, but that seems less ideal.

Either way, we're not accounting for multiple entity types at the moment and I think we need to start doing that.

Reply:
cc: @TimDiekmann @thehabbos007

Reply:
I think having a `link_type: boolean` column makes sense to do for this. 

As for the check, it's unfortunate we can't just use a foreign key, as we would have to conditionally allow for the `entity_type_id` to be a non-link type. I am not sure how idiomatic such a check would be, but if we do it, we definitely need an index on `(version_id, link_type)` to allow the check to be fast as it would be checked in every instance where the entity is moved to the historic table.